File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.cs
using System;
using System.Collections.Generic;

public class Solution {
    public int[] solution(string s) {
        int[] answer = new int[s.Length];
        
        Dictionary<char, int> lastIndex = new Dictionary<char, int>();
        
        for(int i = 0; i < s.Length; i++)
        {
            char c = s[i];
            if(!lastIndex.ContainsKey(c))
            {
                lastIndex[c] = i;
                answer[i] = -1;
            }
            else
            {
                answer[i] = i - lastIndex[c];
            }
            lastIndex[c] = i;
        }
        return answer;
    }
}

File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>

using namespace std;

vector<int> solution(string s) {
    vector<int> answer;
    map<char, int> mp;
    
    for(int i = 0; i < s.size(); i++)
    {
        if(mp.find(s[i]) == mp.end())
        {
            answer.push_back(-1);
        }
        else
        {
            answer.push_back(i - mp[s[i]]);
        }
        mp[s[i]] = i;
    }
    return answer;
}